### ISSUE TITLE
Custom pagers do not work with WSL

### ISSUE THREAD

**Issue by samaursa** [jesseduffield/lazygit#3833]
**Describe the bug**

I have setup diff-so-fancy and it works quite well from the command line in WSL

![image](https://github.com/user-attachments/assets/2453d74d-43d2-4510-a793-7917322f070a)

However, setting the following in `config.yml` has no effect in lazygit

```
git:
  paging:
    colorArg: always
    pager: diff-so-fancy
```

I made sure to check that the `config.yml` is the correct one by customizing branch colours. Which works as expected.

**To Reproduce**

Set a custom pager in lazygit `config.yml` in WSL.

**Expected behavior**

Expect diff-so-fancy to be the pager for lazygit

**Version info:**
commit=71ad3fac63a3ef3326021837b49e9497d332818b, build date=2024-07-13T10:24:19Z, build source=binaryRelease, version=0.43.1, os=windows, arch=amd64, git version=2.41.0.windows.2

**Comment by avario-cpu** [comment]
Works for me on recent Arch WSL with those exact same options in `config.yml`


> git:
>   paging:
>     colorArg: always
>     pager: diff-so-fancy

![Image](https://github.com/user-attachments/assets/eef1ad74-fb1c-4699-a4d8-6ea08754f035)

**Comment by samaursa** [comment]
@avario-cpu Thanks for verifying this. After reading your post, I realized the reason it wasn't working for me was because I was using the Windows version of `lazygit`, which already has this issue reported. After switching to the Linux version, I was able to get the custom pager working as expected.

For anyone curious: the reason I'm using the Windows version of `lazygit` (and `git`) is because my repositories are not located on the WSL filesystem, and accessing them through WSL results in extremely slow git operations (which is a known issue). Closing this issue.

**Comment by stefanhaller** [comment]
@samaursa I'm very inexperienced with WSL, but shouldn't it be possible to set up your PATH in such a way that it finds the Linux version of lazygit, but the Windows version of git?

**Comment by samaursa** [comment]
@stefanhaller I did not know that was a possibility. I am inexperienced with Linux and am unsure how to have Windows git be a system wide (i.e. WSL wide) override. If you are able to help me with setting that up, that would be awesome.

**Comment by samaursa** [comment]
Sadly, it did not work.

Here's the delta pager working correctly in CLI

![Image](https://github.com/user-attachments/assets/10b40b4f-8426-4cfb-b580-a2dd0848a44a)

Here's the config that `lazygit` config that the is installed on `Ubuntu WSL`. The path of the config confirms that it's `lazygit` on linux and not the one on Windows.

![Image](https://github.com/user-attachments/assets/75d2758e-41d4-4ea7-b7c5-e587d7965864)

Here's how I see the diff in `lazygit`

![Image](https://github.com/user-attachments/assets/f43e37eb-18d8-460c-aeb5-e01ae5153bd8)

**Comment by samaursa** [comment]
Another update. Looks like I was still using linux version of git. Here is what happens if I try to launch `lazygit`. Looks like the windows version of `git` is unable to understand the linux paths.

![Image](https://github.com/user-attachments/assets/53398e18-0383-4835-a0fe-d03a96005253)

**Comment by stefanhaller** [comment]
> Here is what happens if I try to launch `lazygit` with Windows version of git. Looks like the windows version of `git` is unable to understand the linux paths.

Yeah, of course, this doesn't work. I didn't think properly, sorry.

It would be useful to make this work, but it would require that lazygit converts the paths somehow. Nothing you can do on your side to make it work, I'm afraid.
